network-providers/basic/interface.md

```diff
@@ -0,0 +1,98 @@
+## IBasicNetworkProvider
+
+```
+interface IBasicNetworkProvider:
+    // Fetches the general configuration of the network.
+    get_network_config(): NetworkConfig;
+
+    // Fetches the current status of the network.
+    get_network_status(shard: number): NetworkStatus;
+
+    // Fetches a block by nonce or by hash.
+    get_block(arguments: GetBlockArguments, url_parameters?: Any): BlockOnNetwork;
+
+    // Fetches the latest block of a shard.
+    get_latest_block(shard: number): BlockOnNetwork;
+
+    // Fetches account information for a given address.
+    // URL parameters can be used, for example, to provide block coordinates for deep-history lookups.
+    get_account(address: IAddress, url_parameters?: Any): AccountOnNetwork;
+
+    // Fetches the storage (key-value pairs) of an account.
+    get_account_storage(address: IAddress, url_parameters?: Any): AccountStorage;
+
+    // Fetches a specific storage entry of an account.
+    get_account_storage_entry(address: IAddress, entry_key: string, url_parameters?: Any): AccountStorageEntry;
+
+    // Waits until an account satisfies a given condition.
+    // Can throw:
+    // - ErrAwaitConditionNotReached
+    await_account_on_condition(
```

Comment:
In constructor, the provided timeout refers to a timeout per request. E.g. timeout for any GET or POST.

However, here, we have a somehow different concept: the timeout in case the condition is not satisfied.